### PR TITLE
Bump to OCaml 4.10.2.

### PR DESCRIPTION
**Pull request by Zimmi48**
Checklist
* [x] Used a fork of the feedstock to propose changes
* (N/A) Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

@conda-forge-admin, please rerender

### CODE REVIEW

**Comment by Zimmi48** [comment]
The Linux error is three tests failing (I don't know how important they are) but the macOS error is an incompatibility with recent clang and gcc versions. We need to bump to OCaml 4.09.1 or later to have a version of OCaml that is compatible. I'll try OCaml 4.10.2 (latest patch-level release---from 3 days ago---for the major version 4.10 which was released in February).

**Comment by Zimmi48** [comment]
@conda-forge-admin, please rerender

**Comment by Zimmi48** [comment]
The patch introduced by @isuruf in #18 does not apply anymore. I didn't look into it yet...

File: recipe/build.sh
Comment (peterjc) [inline recipe/build.sh:8]: This makes sense - you're using the CC, AS and new ASPP environment variables instead, right?

**Comment by peterjc** [comment]
Looks like an OS X test is failing, any idea what beat.ml is?

```
2020-12-16T09:38:27.2483400Z List of failed tests:
2020-12-16T09:38:27.2484140Z     tests/lib-threads/'beat.ml' with 1.2 (native) 
2020-12-16T09:38:27.2484970Z     tests/lib-threads/'beat.ml' with 1.1 (bytecode) 
2020-12-16T09:38:27.2485330Z 
2020-12-16T09:38:27.2485650Z Summary:
2020-12-16T09:38:27.2485950Z   2592 tests passed
2020-12-16T09:38:27.2486280Z    36 tests skipped
2020-12-16T09:38:27.2486600Z     2 tests failed
2020-12-16T09:38:27.2486970Z   103 tests not started (parent test skipped or failed)
2020-12-16T09:38:27.2487380Z     0 unexpected errors
2020-12-16T09:38:27.2487710Z   2733 tests considered
2020-12-16T09:38:27.2488100Z #### Something failed. Exiting with error status.
```

File: recipe/build.sh
Comment (Zimmi48) [reply recipe/build.sh:8]: Yes. These options are not supported anymore apparently, but `./configure --help` says that the environment variables that you mentioned are taken into account.

Comment (peterjc) [reply recipe/build.sh:8]: So that change is prompted by updating from 4.06.1 to 4.10.2 - understood, thanks.

**Comment by Zimmi48** [comment]
The test file is here: https://github.com/ocaml/ocaml/blob/trunk/testsuite/tests/lib-threads/beat.ml

I have no idea if this failing test should be a worry or not. I noticed that the CI of the dev repo of OCaml doesn't include testing on macOS, so it's a bit difficult to know if it's working or not on this platform.

Looking into this error on Google, all I found is this: https://github.com/ocaml-multicore/ocaml-multicore/issues/408
It's hard to know if this failure is specific to the multi-core repo or not though.

**Comment by peterjc** [comment]
Sounds like a known failure on macOS - do you think as a quick and dirty workaround we can delete that file to skip the test?

**Comment by Zimmi48** [comment]
Yes, I agree. Done.

**Review by peterjc — APPROVED** [review]

**Comment by Zimmi48** [comment]
@peterjc I'm waiting for you to merge this to be able to update the coq-feedstock version.

**Comment by peterjc** [comment]
Merged. Sorry, I forgot I was still the sole named maintainer on this - happy to add you if you would like? As I have said in other comments on this receipt, I am not an Ocaml user myself and just wanted this in conda as a dependency of something else.

**Comment by Zimmi48** [comment]
Feel free to add me. I'm an OCaml user myself though not an actual conda user (:laughing:). I just help maintain the Coq recipe and I might as well do the same for OCaml.

**Pull request merged**
